core/new-gui/src/app/dashboard/component/feature-container/user-project-list/user-project-list.component.ts

```diff
@@ -0,0 +1,134 @@
+import { Component, OnInit } from "@angular/core";
+import { UserProjectService } from "../../../service/user-project/user-project.service";
+import { UserProject } from "../../../type/user-project";
+import { Router } from "@angular/router";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { NotificationService } from "../../../../common/service/notification/notification.service";
+
+export const ROUTER_USER_PROJECT_BASE_URL = "/dashboard/user-project";
+
+@UntilDestroy()
+@Component({
+  selector: "texera-user-project-list",
+  templateUrl: "./user-project-list.component.html",
+  styleUrls: ["./user-project-list.component.scss"]
+})
+export class UserProjectListComponent implements OnInit {
+  public userProjectEntries: UserProject[] = [];
+  public userProjectEntriesIsEditingName: number[] = [];
+  public createButtonIsClicked: boolean = false;
+  public createProjectName: string = "";
+
+  constructor(
+    private userProjectService: UserProjectService,
+    private router: Router,
+    private notificationService: NotificationService
+  ) { 
+  }
+
+  ngOnInit(): void {
+    this.getUserProjectArray();
+  }
+
+  private getUserProjectArray() {
+    
+    this.userProjectService
+      .retrieveProjectList()
+      .pipe(untilDestroyed(this))
+      .subscribe(projectEntries => {
+        this.userProjectEntries = projectEntries;
+      });
+  }
+
+  /**
+   * navigate to individual project page
+   */
+  public jumpToProject({ pid }: UserProject): void {
+    this.router.navigate([`${ROUTER_USER_PROJECT_BASE_URL}/${pid}`]).then(null);
+  }
+
+  public removeEditStatus(pid : number): void {
+    this.userProjectEntriesIsEditingName = this.userProjectEntriesIsEditingName.filter(index => index != pid);
+  }
+
+  public saveProjectName(project: UserProject, newName: string, index: number): void {
+    // nothing happens if name is the same
+    if (project.name === newName) {
+      this.removeEditStatus(project.pid);
+    } else if (this.isValidNewProjectName(newName, project)) {
+      this.userProjectService
+      .updateProjectName(project.pid, newName)
+      .pipe(untilDestroyed(this))
+      .subscribe(() => {
+        this.removeEditStatus(project.pid);
+        this.getUserProjectArray(); // refresh list of projects, name is read only property so can"t edit
+      });
```

Comment:
when backend reject this call (return non-200), where is the handler?

Reply:
backend will throw BadRequestExceptions for invalid names creating / renaming projects, frontend code for creating / renaming project now updated to have error handlers